### PR TITLE
Adds initial pass at preflight.

### PR DESCRIPTION
**Pull request by kristoferjoseph**

### CODE REVIEW

File: src/http/any-catchall/_get-preflight.mjs
Comment (tbeseda) [inline src/http/any-catchall/_get-preflight.mjs:1]: Could we check if it exists, then import and execute without a catch? that way my app's code will throw and I notice, get an error/stack, fix it

File: src/http/any-catchall/router.mjs [hunk 1]
```diff
@@ -141,9 +143,7 @@ export default async function api(options, req) {
   let elements = { ...altHeadElements.elements, ...baseHeadElements.elements }
   timers.stop('elements')
 
-  const store = state.json
-    ? state.json
-    : {}
+  const store = Object.assign(preflight(req), state.json ? state.json : {})
```
[the remark this answers is not in the document]
Comment (kristoferjoseph) [reply src/http/any-catchall/router.mjs:146]: That is a merge issue. Will fix

File: src/http/any-catchall/_get-preflight.mjs
Comment (kristoferjoseph) [reply src/http/any-catchall/_get-preflight.mjs:1]: We have to try catch the import but i want to throw on anything but file not found

**Comment by kristoferjoseph** [comment]
This feels like the right way to do preflight.
It checks to see if the file exists before importing so that if there are any user errors in the file itself there will be a useful error thrown.

File: test/_get-preflight.mjs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  t.plan(1)
+  const basePath = path.join(__dirname, 'mock-app', 'app')
+  const preflight = await getPreflight({ basePath })
+  t.notok(preflight, 'Missing preflight is OK')
+})
+/*
+// No idea how to get this type of test to actually work as documented. 
+// Maybe someone with more patience can get this to do it's one job
+// This works as expected and if this type of test actually worked it would prove it
+test('preflight with missing module import inside it should throw', async t => {
```
Comment (tbeseda) [inline test/_get-preflight.mjs:31]: re: tape's `throws` with async functions, I usually end up with something like this

```js
test('preflight with missing module import inside it should throw', async t => {
  t.plan(1)
  const basePath = path.join(__dirname, 'mock-preflight', 'app', 'bad-preflight')
  // t.throws(async function () { await getPreflight({ basePath }) })
  try {
    await getPreflight({ basePath })
    t.fail('Missing module import should throw')
  }
  catch (error) {
    t.ok(error, 'Missing module import throws')
  }
})
```

Comment (kristoferjoseph) [reply test/_get-preflight.mjs:31]: Awesome thank you! I’ll add this test

**Review by macdonst — APPROVED** [review]

**Review by colepeters — APPROVED** [review]
Nice one! 🤩

**Comment by kristoferjoseph** [comment]
https://github.com/enhance-dev/preflight-demo

**Pull request merged**